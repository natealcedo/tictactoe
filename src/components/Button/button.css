.button-row {
  margin: 0 auto;
}

.button {
  border: 1px solid #c0c0c0;
  height: 50px;
  width: 300px;
  text-align: center;
  border-radius: 5px;
  background-color: #00363d;
  color: white;
  display: flex;
  margin: auto;
  margin-bottom: 5px;
}

.button:hover {
  cursor: pointer;
}

.button > div {
  margin: auto;
}