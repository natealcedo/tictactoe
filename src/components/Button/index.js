import React from "react";
import propTypes from "prop-types";
import "./button.css";

const Button = ({ label, onClick }) => (
  <div className="button" onClick={() => onClick()}>
    <div>{label}</div>
  </div>
);

Button.propTypes = {
  label: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired
};

export default Button;
