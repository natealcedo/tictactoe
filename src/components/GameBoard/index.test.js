import React from "react";
import { GameBoard } from ".";
import { shallow } from "enzyme";
import Button from "../Button";

describe("GameBoard", () => {
  const noOp = () => {};
  test("should render without exploding", () => {
    const gameState = [[0, 0, 1], [1, 1, 0], [0, 1, 0]];
    const currentPlayer = {
      name: "Nate",
      numberOfMoves: 5,
      valueOnBoard: 0
    };
    const wrapper = shallow(
      <GameBoard
        currentPlayer={currentPlayer}
        gameState={gameState}
        isDraw={false}
        isGameStarted={true}
        resetGame={noOp}
        startNewGame={noOp}
        takeTurn={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("should show drawn", () => {
    const gameState = [[1, 0, 0], [0, 1, 1], [1, 1, 0]];
    const currentPlayer = {
      name: "Nate",
      numberOfMoves: 5,
      valueOnBoard: 1
    };
    const wrapper = shallow(
      <GameBoard
        currentPlayer={currentPlayer}
        gameState={gameState}
        isDraw={true}
        isGameStarted={true}
        resetGame={noOp}
        startNewGame={noOp}
        takeTurn={() => {}}
      />
    );
    expect(wrapper.find(".state-tracker > h2")).toIncludeText("Game Drawn");
    expect(wrapper).toMatchSnapshot();
  });

  test('should show "Nate won"', () => {
    const gameState = [[1, 0, 0], [1, 1, 1], [0, 1, 0]];
    const currentPlayer = {
      name: "Nate",
      numberOfMoves: 5,
      valueOnBoard: 1
    };
    const wrapper = shallow(
      <GameBoard
        currentPlayer={currentPlayer}
        gameState={gameState}
        isDraw={false}
        isGameStarted={true}
        resetGame={noOp}
        startNewGame={noOp}
        takeTurn={() => {}}
        winner="player2"
      />
    );
    expect(wrapper.find(".state-tracker > h2")).toIncludeText("NATE won!");
    expect(wrapper).toMatchSnapshot();
  });

  test("should call resetGame and startNewGame", () => {
    const fn = jest.fn();
    const gameState = [[1, 0, 0], [1, 1, 1], [0, 1, 0]];
    const currentPlayer = {
      name: "Nate",
      numberOfMoves: 5,
      valueOnBoard: 1
    };
    const wrapper = shallow(
      <GameBoard
        currentPlayer={currentPlayer}
        gameState={gameState}
        isDraw={false}
        isGameStarted={true}
        resetGame={fn}
        startNewGame={fn}
        takeTurn={() => {}}
        winner="player2"
      />
    );
    wrapper
      .find(Button)
      .at(0)
      .simulate("click");
    wrapper
      .find(Button)
      .at(1)
      .simulate("click");
    expect(fn).toHaveBeenCalledTimes(2);
    expect(wrapper).toMatchSnapshot();
  });

  test("should not render ", () => {
    const gameState = [
      [null, null, null],
      [null, null, null],
      [null, null, null]
    ];
    const wrapper = shallow(
      <GameBoard
        gameState={gameState}
        isDraw={false}
        isGameStarted={false}
        resetGame={noOp}
        startNewGame={noOp}
      />
    );
    expect(wrapper).toBeEmptyRender();
  });
});
