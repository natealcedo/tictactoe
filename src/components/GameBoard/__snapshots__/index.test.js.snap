// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard should render without exploding 1`] = `
<div
  className="game-board-container"
>
  <div
    className="state-tracker"
  >
    <h2>
      NATE's turn
    </h2>
  </div>
  <div
    className="game-board"
  >
    <Square
      column={0}
      key="0-0"
      onClick={[Function]}
      row={0}
      value={0}
    />
    <Square
      column={1}
      key="0-1"
      onClick={[Function]}
      row={0}
      value={0}
    />
    <Square
      column={2}
      key="0-2"
      onClick={[Function]}
      row={0}
      value={1}
    />
    <Square
      column={0}
      key="1-0"
      onClick={[Function]}
      row={1}
      value={1}
    />
    <Square
      column={1}
      key="1-1"
      onClick={[Function]}
      row={1}
      value={1}
    />
    <Square
      column={2}
      key="1-2"
      onClick={[Function]}
      row={1}
      value={0}
    />
    <Square
      column={0}
      key="2-0"
      onClick={[Function]}
      row={2}
      value={0}
    />
    <Square
      column={1}
      key="2-1"
      onClick={[Function]}
      row={2}
      value={1}
    />
    <Square
      column={2}
      key="2-2"
      onClick={[Function]}
      row={2}
      value={0}
    />
  </div>
</div>
`;
